src/Microsoft.IdentityModel.JsonWebTokens/JwtTokenUtilities.cs

```diff
@@ -380,9 +381,53 @@ private static long ParseTimeValue(JToken jToken, string claimName)
         /// <param name="validationParameters">A <see cref="TokenValidationParameters"/>  required for validation.</param>
         /// <returns>Returns a <see cref="SecurityKey"/> to use for signature validation.</returns>
         /// <remarks>If key fails to resolve, then null is returned</remarks>
-        internal static SecurityKey ResolveTokenSigningKey(string kid, string x5t, TokenValidationParameters validationParameters)
+        internal static SecurityKey ResolveTokenSigningKeyWithConfig(string kid, string x5t, TokenValidationParameters validationParameters)
         {
+            StandardConfiguration configuration = null;
+            if (validationParameters.ConfigurationManager != null)
+                configuration = validationParameters.ConfigurationManager.GetStandardConfigurationAsync(CancellationToken.None).Result;
```

Comment:
This needs to be synchronous for now as we do not want to add a ValidateAsync(...) API that doesn't have asynchronous verifying operations.